src/test/regress/spec/isolation_logical_replication_nonsu_nonbypassrls.spec

```diff
@@ -0,0 +1,95 @@
+// test moving a single shard that has rls
+// owned by a user that is neither superuser nor bypassrls
+// PG15 added extra permission checks within logical replication
+// this test makes sure that target table owners should still
+// be able to replicate despite RLS policies.
+setup
+{
+	SET citus.shard_count TO 4;
+	SET citus.shard_replication_factor TO 1;
+
+    CREATE USER new_user;
+    GRANT ALL ON SCHEMA public TO new_user;
+
+    CREATE TABLE dist(column1 int PRIMARY KEY, column2 int);
+    SELECT create_distributed_table('dist', 'column1');
+
+	SELECT get_shard_id_for_distribution_column('dist', 23) INTO selected_shard;
+    GRANT ALL ON TABLE selected_shard TO new_user;
+}
+
+teardown
+{
+    DROP TABLE selected_shard;
+    DROP TABLE dist;
+    REVOKE ALL ON SCHEMA public FROM new_user;
+    DROP USER new_user;
+}
+
+session "s1"
+
+step "s1-table-owner-new_user"
+{
+    ALTER TABLE dist OWNER TO new_user;
+}
+
+step "s1-table-enable-rls"
+{
+    ALTER TABLE dist ENABLE ROW LEVEL SECURITY;
```

Comment:
Can you also try by adding: 
```
ALTERT TABLE dist FORCE ROW LEVEL SECURITY; 
```

I think that might actually fail.

Reply:
Actually, it doesn't fail, because of this:

https://github.com/postgres/postgres/blob/af51b2f042a0808943125b82c0d6e785259069f2/src/backend/utils/misc/rls.c#L99-L118